flytekit/models/sagemaker/parameter_ranges.py

```diff
@@ -177,11 +178,17 @@ def to_flyte_idl(self) -> _idl_parameter_ranges.ParameterRanges:
     def from_flyte_idl(cls, pb2_object: _idl_parameter_ranges.ParameterRanges):
         converted = {}
         for k, v in pb2_object.parameter_range_map.items():
-            if isinstance(v, _idl_parameter_ranges.ContinuousParameterRange):
-                converted[k] = ContinuousParameterRange.from_flyte_idl(v)
-            elif isinstance(v, _idl_parameter_ranges.IntegerParameterRange):
-                converted[k] = IntegerParameterRange.from_flyte_idl(v)
+            # Use WhichOneof to find out which field in the oneof has been set
+            # https://developers.google.com/protocol-buffers/docs/reference/python-generated#oneof
+            # and then use getattr to get the field, and use isinstance to check if that field is indeed that type
+            # https://stackoverflow.com/questions/51421392/dynamically-access-oneof-value-from-a-protobuf
+            if isinstance(getattr(v, v.WhichOneof('parameter_range_type')),
```

Comment:
I thought we usually used HasField for this stuff.

Reply:
fixed :)